### PR TITLE
configurble metrics, fix #1338

### LINKED ISSUE #1338: Make Monitoring facility switchable

### PR DESCRIPTION
**Pull request by zambrovski**
### Description

Short description or comments

### Reference

Issues: closes #1338

### Check-List

- [ ] All Acceptance criteria of user story are met
- [ ] Accessibility was considered and tested (On UI Change)
- [ ] JUnit tests are written (60% CodeCov)
- [ ] [Internal Review]([https://github.com/it-at-m/digiwf-core/blob/dev/CHANGELOG.md](https://confluence.muenchen.de/display/MPdZ/Review+-+DigiWF)) is maintained
- [ ] Documentations [external](https://github.com/it-at-m/digiwf-core/tree/dev/docs) and [internal](https://wiki.muenchen.de/betriebshandbuch/index.php?title=DigiWF&sfr=betriebshandbuch) are completed
- [ ] Smoketest successful (Manual E2E-Test depending on Change) 
- [ ] No waste on Branch left (e.g. `console.logs`)
- [ ] [Board](https://app.zenhub.com/workspaces/digiwf-621f70bf50ea1100120b7e93/board) is up-to-date
- [ ] Openshift environments are prepared (Secrets, etc.) and release-issue is maintained

### CODE REVIEW

File: digiwf-libs/digiwf-camunda-prometheus/digiwf-camunda-prometheus-starter/src/main/java/de/muenchen/oss/digiwf/camunda/prometheus/MetricsProviderSchedulerAutoConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    private final CollectorRegistry collectorRegistry;
+
+    @PostConstruct
+    public void initializeProviderMetrics() {
+        // providers
+        log.info("[DIGIWF METRICS]: Registered {} metrics providers: {}", metricsProviders.keySet().size(), String.join(", ", metricsProviders.keySet()));
+        metricsProviders.forEach((key, value) -> value.registerMetrics(collectorRegistry));
+    }
+
+    @Scheduled(fixedDelayString = "${digiwf.prometheus.process-engine.update-interval:3000}")
```
Comment (simonhir) [inline digiwf-libs/digiwf-camunda-prometheus/digiwf-camunda-prometheus-starter/src/main/java/de/muenchen/oss/digiwf/camunda/prometheus/MetricsProviderSchedulerAutoConfiguration.java]:
```suggestion
    @Scheduled(fixedDelayString = "${digiwf.prometheus.process-engine.update-interval}")
```

Verschoben in Configuration

File: digiwf-libs/digiwf-camunda-prometheus/digiwf-camunda-prometheus-example/src/main/resources/application.yaml [hunk 2]
```diff
@@ -1,8 +1,13 @@
-io:
-  muenchendigital:
-    camunda:
-      prometheus:
-        update-interval: 3000
+digiwf:
+  prometheus:
+    process-engine:
+      update-interval: 3000
```
Comment (simonhir) [inline digiwf-libs/digiwf-camunda-prometheus/digiwf-camunda-prometheus-example/src/main/resources/application.yaml]:
```suggestion
      update-interval: 30000
```

File: digiwf-engine/digiwf-engine-service/src/main/resources/application.yml [hunk 3]
```diff
@@ -185,6 +185,16 @@ digiwf:
   s3service:
     topic: 'dwf-s3-${DIGIWF_ENV}'
     httpAPI: ${DIGIWF_S3_HTTPAPI:http://localhost:8086}
+  prometheus:
+    process-engine:
+      update-interval: 3000
```
Comment (simonhir) [inline digiwf-engine/digiwf-engine-service/src/main/resources/application.yml]:
```suggestion
      update-interval: 30000
```

Sind aktuell 3000ms -> 3s wenn ich die Doku richtig gelesen habe

File: digiwf-libs/digiwf-camunda-prometheus/digiwf-camunda-prometheus-starter/src/main/java/de/muenchen/oss/digiwf/camunda/prometheus/CamundaPrometheusProperties.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
+package de.muenchen.oss.digiwf.camunda.prometheus;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+@ConfigurationProperties(prefix = "digiwf.prometheus.process-engine")
+public class CamundaPrometheusProperties {
+    /**
+     * Update interval for running metrics providers, which actively execute queries.
+     */
+    private int updateInterval;
```
Comment (simonhir) [inline digiwf-libs/digiwf-camunda-prometheus/digiwf-camunda-prometheus-starter/src/main/java/de/muenchen/oss/digiwf/camunda/prometheus/CamundaPrometheusProperties.java]:
```suggestion
    private int updateInterval = 30000;
```

Würde den Default jetzt eher über die Config setzen.

File: digiwf-libs/digiwf-camunda-prometheus/digiwf-camunda-prometheus-example/src/main/resources/application.yaml (hunk 2 above)
Comment (zambrovski) [reply digiwf-libs/digiwf-camunda-prometheus/digiwf-camunda-prometheus-example/src/main/resources/application.yaml]: das ist ein Example... Keine echte Konfig

File: digiwf-engine/digiwf-engine-service/src/main/resources/application.yml (hunk 3 above)
Comment (zambrovski) [reply digiwf-engine/digiwf-engine-service/src/main/resources/application.yml]: das ist die echte Engine. Ok aus meiner Sicht.

File: digiwf-libs/digiwf-camunda-prometheus/digiwf-camunda-prometheus-example/src/main/resources/application.yaml (hunk 2 above)
Comment (simonhir) [reply digiwf-libs/digiwf-camunda-prometheus/digiwf-camunda-prometheus-example/src/main/resources/application.yaml]: Wollte es nur überall gleich ziehen

**Review by simonhir — APPROVED** [review]

**Pull request merged**